website_forum_censored/models/forum_post.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+# © 2016 Onestein (<http://www.onestein.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import api, models
+from re import compile, IGNORECASE
+
+
+class ForumPost(models.Model):
+    _inherit = 'forum.post'
+
+    @api.model
+    def create(self, values):
+        phrases = self.env['forum.censored.phrase'].search([])
+        for phrase in phrases:
+            ptn = compile(phrase.phrase, IGNORECASE)
+            if 'name' in values:
+                values['name'] = ptn.sub(phrase.replacement, values['name'])
+            if 'content' in values:
+                values['content'] = ptn.sub(phrase.replacement,
+                                            values['content'])
```

Comment:
from a stylistic point of view, I'd have a method `replace()` on the `forum.censored.phrase` model.